packages/server/lib/cloud/protocol.ts

```diff
@@ -344,7 +395,7 @@ export class ProtocolManager implements ProtocolManagerShape {
    * Abstracts invoking a synchronous method on the AppCaptureProtocol instance, so we can handle
    * errors in a uniform way
    */
-  private async invokeAsync <K extends ProtocolAsyncMethods> (method: K, ...args: Parameters<AppCaptureProtocolInterface[K]>) {
+  private async invokeAsync <K extends ProtocolAsyncMethods> (method: K, isEssential: boolean, ...args: Parameters<AppCaptureProtocolInterface[K]>) {
```

Comment:
can we make the second option an object to make the intention of the boolean a bit clearer without necessarily having to look up the method? like:

```
this.invokeAsync('beforeTest', { isEssential: true }, test)
```

Reply:
Considered it, and thought vscode's hinting would be enough. I'll change it up.